statsforecast/core.py

```diff
@@ -907,49 +626,37 @@ def _make_future_df(self, h: int):
             dates = np.tile(last_date_f(self.last_dates[0]), len(self.ga))
         else:
             dates = np.hstack([last_date_f(last_date) for last_date in self.last_dates])
-        u_id_ser: Union[pd.Series, pl.Series] = np.repeat(self.uids, h)
-        unique_id: np.ndarray = u_id_ser.to_numpy()
-
-        # In older versions to_numpy converts string values into object,
-        # creating bytes error, this fixes it
-        if unique_id.dtype.kind == "O":
-            unique_id = unique_id.astype(str)
-
-        if self.engine == pd.DataFrame:
-            idx = pd.Index(unique_id, name="unique_id")
-            df = self.engine({"ds": dates}, index=idx)
-        elif self.engine == pl.DataFrame:
-            df = self.engine({"unique_id": unique_id, "ds": dates})
-        return df
 
-    def _parse_X_level(self, h, X, level):
-        if X is not None:
-            if isinstance(X, pd.DataFrame):
-                if X.index.name != "unique_id":
-                    X = X.set_index("unique_id")
-            expected_shape_rows = h * len(self.ga)
-            ga_shape = self.ga.data.shape[1]
-            # Polars doesn't have index, hence, extra "column"
-            expected_shape_cols = (
-                ga_shape if not isinstance(X, pl.DataFrame) else ga_shape + 1
-            )
-            expected_shape = (expected_shape_rows, expected_shape_cols)
+        uids = np.repeat(self.uids, h)
+        if self.df_constructor is pl_DataFrame and uids.dtype.kind == "O":
+            uids = uids.astype(str)
+        df = self.df_constructor({"unique_id": uids, "ds": dates})
+        if isinstance(df, pd.DataFrame):
+            df = df.set_index("unique_id")
```

Comment:
for the future: should we deprecate returning an indexed data frame in the future? it would be a breaking change but more compatible with the nixtlaverse, wdyt?